build.sbt

```diff
@@ -1,58 +1,77 @@
+  "-language:higherKinds",
+  "-language:implicitConversions",
+  "-language:postfixOps",
+  "-unchecked",
+  "-Xfuture",
+  "-Xlint",
+  "-Yno-adapted-args",
+  "-Ypartial-unification",
+  "-Ywarn-extra-implicit",
+  "-Ywarn-inaccessible",
+  "-Ywarn-infer-any",
+  "-Ywarn-nullary-override",
+  "-Ywarn-nullary-unit",
+  "-Ywarn-numeric-widen",
+  "-Ywarn-unused:implicits",
+  "-Ywarn-unused:imports",
+  "-Ywarn-unused:locals",
+  "-Ywarn-unused:patvars",
+  "-Ywarn-unused:privates",
+  "-Ywarn-value-discard",
+) ++ {
+  if (scala.sys.env.getOrElse("CI", "") == "true") {
+    Seq("-Xfatal-warnings")
+  } else { Seq() }
+}
+
+lazy val graphientCore = (project in file("graphient"))
   .settings(
     name := "graphient",
     scalaVersion := "2.12.6",
-    version := "4.0.1",
-    scalacOptions ++= Seq(
-      "-deprecation",
-      "-encoding",
-      "utf-8",
-      "-explaintypes",
-      "-feature",
-      "-language:existentials",
-      "-language:higherKinds",
-      "-language:implicitConversions",
-      "-language:postfixOps",
-      "-unchecked",
-      "-Xfuture",
-      "-Xlint",
-      "-Yno-adapted-args",
-      "-Ypartial-unification",
-      "-Ywarn-extra-implicit",
-      "-Ywarn-inaccessible",
-      "-Ywarn-infer-any",
-      "-Ywarn-nullary-override",
-      "-Ywarn-nullary-unit",
-      "-Ywarn-numeric-widen",
-      "-Ywarn-unused:implicits",
-      "-Ywarn-unused:imports",
-      "-Ywarn-unused:locals",
-      "-Ywarn-unused:patvars",
-      "-Ywarn-unused:privates",
-      "-Ywarn-value-discard",
-    ),
-    scalacOptions ++= {
-      if (scala.sys.env.getOrElse("CI", "false") == "true")  {
-        Seq("-Xfatal-warnings")
-      }
-      else {
-        Seq()
-      }
-    },
+    version := "5.0.0-SNAPSHOT",
+    scalacOptions ++= ScalacOptions,
     resolvers += Resolver.sonatypeRepo("releases"),
     bintrayRepository := "io.github.erdeszt",
     organization := "io.github.erdeszt",
     licenses += ("MIT", url("http://opensource.org/licenses/MIT")),
     addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.7"),
     libraryDependencies ++= sangria,
-    libraryDependencies ++= circe,
     libraryDependencies ++= cats,
     libraryDependencies ++= sttp,
     libraryDependencies ++= scalaTest.map(_ % Test),
-    libraryDependencies ++= circeTest,
-    libraryDependencies ++= http4s.map(_ % Test)
   )
 
-lazy val graphientProject = (project in file(".")).aggregate(graphient)
+lazy val graphientCirce = (project in file("graphient-circe"))
+  .settings(
+    name := "graphient-circe",
+    scalaVersion := "2.12.6",
+    version := "1.0.0",
+    scalacOptions ++= ScalacOptions,
+    resolvers += Resolver.sonatypeRepo("releases"),
+    licenses += ("MIT", url("http://opensource.org/licenses/MIT")),
+    libraryDependencies ++= circe,
+    libraryDependencies ++= scalaTest.map(_ % Test),
+  ).dependsOn(graphientCore % "compile->compile;test->test")
+
+lazy val graphientSpray = (project in file("graphient-spray"))
```

Comment:
I needs an organization `organization := "io.github.erdeszt",` to get the publish work - I tried out locally with `sbt graphientSpray/publishLocal`

and without organization i published it to the wrong spot:

`/home/pelle/.ivy2/local/graphient-spray/graphient-spray_2.12/1.0.0-SNAPSHOT/ivys/ivy.xml`

consumers look for it here

`/home/pelle/.ivy2/local/io.github.erdeszt/graphient-spray_2.12/1.0.0-SNAPSHOT/ivys/ivy.xml`